### DISCUSSION TITLE
VGA Question : read back in console mode ?

### DISCUSSION THREAD

**Discussion by GlennFR** [picocomputer/community#113] [Q&A]
Hello. I'm waiting for my picocomputer boards to arrive from China, meanwhile I'm reading-up all I can about the system.  I have a question about the VGA engine.  I saw the "Game of life" YT video, and I want to know if, in console mode, the same readback mechanism will return the chrs present in 80-column console mode.
Thank for your inputs.
Well, a long time later - I've unmarked the answer because it **isn't** really the answer to my question (the said editor is using the RIA X-RAM, not the VGA ram)...
I'll try to be clearer with the question : I wanted to know if it's possible to read back from the VGA console memory, as it used to be possible in the old days of the VT100 and other "real" terminals.  An equivalent of reading back from video memory in "Text mode" on most original 6502 platforms (Apples, Acorns, Commodores, etc)

**Comment by rumbledethumps** [comment]
It sounds like you're asking for VGA mode 1. Or maybe mode 2. It's not clear what you're trying to do so providing guidance is nigh impossible.

**Reply by GlennFR** [reply]
Hi, thanks for your reply.  For the moment it's just curiosity and trying to completely understand the system.  I was wondering if, when using the 80-column text console, if it's possible to read back the chr. codes from the video memory.  I was musing about an editor that worked directly from the video memory.
Thanks again.  Sill patiently waiting for my boards...

**Reply by rumbledethumps** [reply]
Here's a text editor you can study or use: https://github.com/orgs/picocomputer/discussions/96

**Reply by GlennFR** [reply]
Oops, messed-up in writing my reply.  So, yes, I see in the text editor example that it's doing pretty-much exactly what I was imagining.  Thanks.  So, I'm losing patience waiting for my boards...

**Reply by GlennFR** [reply]
Well, a long time later - I've unmarked the answer because it **isn't** really the answer to my question (the said editor is using the RIA X-RAM, not the VGA ram)...
I'll try to be clearer with the question : I wanted to know if it's possible to read back from the VGA console memory, as it used to be possible in the old days of the VT100 and other "real" terminals.  An equivalent of reading back from video memory in "Text mode" on most original 6502 platforms (Apples, Acorns, Commodores, etc)

**Reply by rumbledethumps** [reply]
> I wanted to know if it's possible to read back from the VGA console memory, as it used to be possible in the old days of the VT100 and other "real" terminals. 

If this is true then why can't we do it on a modern system with terminfo?

**Reply by GlennFR** [reply]
Hi, Rumble. Your reply is not really useful.  I was an engineer for HP on the HP1000 series minicomputers.  If you sent "[ESC]&p" to any of the HP26xx series terminals, they would send you the current line from the current cursor.  The HP1000 had a brilliant editor that exploited this sequence.  Unfortunately the HP codes were independent of the DEC codes that were adopted for the ANSI standard. At the time I had written some code to interface to a DEC-compatible terminal - the DEC codes couldn't send a line, but *could* send chr at cursor.  One of the revisions of the ANSI standard removed such sequences because they were considered a security risk since some systems were being accessed remotely and not through direct wired connections.  However, my stance is much more the retro computer form where it was common practice to PEEK into video memory to see what chrs were there.
NYWAY, can you just reply with YES it can be done, or NO it can't be done?  Would save a lot of time for everyone. Thanks.
